diff --git a/node_modules/react-native-verus/scripts/copyCheckpoints.ts b/node_modules/react-native-verus/scripts/copyCheckpoints.ts
--- a/node_modules/react-native-verus/scripts/copyCheckpoints.ts
+++ b/node_modules/react-native-verus/scripts/copyCheckpoints.ts
@@ -7,7 +7,7 @@ export async function copyCheckpoints(disklet: Disklet): Promise<void> {
   console.log('Copying checkpoints...')
   const fromDisklet = navigateDisklet(
     disklet,
-    'android/src/main/assets/co.electriccoin.zcash/checkpoint/vrsc'
+    'android/src/main/assets/co.electriccoin.zcash/checkpoint/mainnet'
   )
   const toDisklet = navigateDisklet(
     disklet,
